SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::AsyncGenerateDevAuthToken()
+{
+	bool bExpected = false;
+	if (bIsGenerating.CompareExchange(bExpected, true))
+	{
+		bool bIsRunningInChina = GetMutableDefault<USpatialGDKSettings>()->IsRunningInChina();
```

Comment:
```suggestion
		bool bIsRunningInChina = GetDefault<USpatialGDKSettings>()->IsRunningInChina();
```

no need for a mutable object afaics